packages/agoric-cli/src/install.js

```diff
@@ -134,21 +136,72 @@ export default async function installMain(progname, rawArgs, powers, opts) {
         if (!packageJSON) {
           return;
         }
+        const obsolete = new Set();
         const pj = JSON.parse(packageJSON);
         for (const section of ['dependencies', 'devDependencies']) {
           const deps = pj[section];
           if (deps) {
             for (const pkg of Object.keys(deps)) {
               if (versions.has(pkg)) {
+                if (deps[pkg] === forceSdkVersion) {
```

Comment:
I would have thought we need to "invalidate the cache" on a version _mismatch_.

I can't see where `deps` is initialized in this diff, and when I looked at the whole file, I got a little dizzy. (color me lazy)

Reply:
> I would have thought we need to "invalidate the cache" on a version mismatch.

Actually, `yarn` already updates packages that have changed.  It's the packages whose version specifier has *not* changed that we need to first prune the old one, install, add the new one, and install to get it to update correctly.